packages/validation/src/isAsync.js

```diff
@@ -0,0 +1,9 @@
+import isPromise from './isPromise';
+import isFunction from './isFunction';
+
+export default function isAsync(value) {
+    return (
+        isPromise(value) ||
+        (isFunction(value) && /return new Promise|asyncGeneratorStep/.test(value.toString().trim()))
```

Comment:
```js
const AsyncFunction = (async () => {}).constructor;
const GeneratorFunction = (function* () => {}).constructor;

const isAsync = (value) => {
return isPromise(value) || 
    (isFunction(value) && value.constructor.name === 'AsyncFunction')  ||  // native 
    (value instanceof AsyncFunction && AsyncFunction !== Function && AsyncFunction !== GeneratorFunction);
};
```

Reply:
false for every value....

Reply:
you have to include `isFunction(value) && value.constructor.name === 'AsyncFunction'` because this function wont run always as a compiled version by babel, in fact in nodejs will return false if you don't include it